test/kuzzle/query.test.js

```diff
@@ -242,5 +242,14 @@ describe('Kuzzle query management', () => {
 
       should(request).be.eql({ controller: 'server', action: 'now' });
     });
+
+    it('should call logDeprecation with the response', async () => {
+      kuzzle.deprecationHandler.logDeprecation = sinon.stub().returns(response);
+      await kuzzle.query(query);
+
+      should(kuzzle.deprecationHandler.logDeprecation)
+        .be.calledOnce()
```

Comment:
Nitpicking: this line is useless since you check that the method is called with the response

Reply:
But I make myself sure that the function is called only once, I feel like I'm also nitpicking!